lib/plugins/threaded-signing/plugin.c

```diff
@@ -254,31 +280,33 @@ threaded_openssl_get_signature(sv_threaded_plugin_t *self,
   SignedVideoReturnCode status = SV_OK;
 
   g_mutex_lock(&self->mutex);
-  if (self->plugin_state == THREADED_SIGNING_HAS_SIGNATURE && self->signature_info) {
-    if (self->signature_info->signature_size > max_signature_size) {
-      // If there is no room to copy the signature, report zero size.
-      *written_signature_size = 0;
-    } else {
-      memcpy(signature, self->signature_info->signature, self->signature_info->signature_size);
-      *written_signature_size = self->signature_info->signature_size;
-    }
-    // Change state and mark as copied.
-    self->plugin_state = THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN;
-    has_copied_signature = true;
-  } else if (self->plugin_state == THREADED_SIGNING_ERROR) {
-    *written_signature_size = 0;
-    // Change state and mark as copied.
-    self->plugin_state = THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN;
-    has_copied_signature = true;
+  if (self->output_buffer_idx == 0) goto done;
+  *written_signature_size = 0;
```

Comment:
Add empty line before this one to avoid associating it with the if-statement above